client/my-sites/checkout/checkout/composite-checkout.js

```diff
@@ -612,39 +612,40 @@ export default function CompositeCheckout( {
 	const validateDomainContact =
 		validateDomainContactDetails || wpcomValidateDomainContactInformation;
 
-	const renderDomainContactFields = (
+	const domainContactValidationCallback = (
+		paymentMethodId,
+		contactDetails,
 		domainNames,
+		applyDomainContactValidationResults
+	) => {
+		return validateDomainContact( contactDetails, domainNames, ( httpErrors, data ) => {
```

Comment:
Below, when `domainContactValidationCallback` is being called, it has an `await` before it. However, I'm pretty sure that this function does not return a Promise. The default, `Undocumented.prototype.validateDomainContactInformation`, passes a callback argument to `wpcom.req.post` which I believe makes it use the callback instead of a Promise.

Since we probably need to wait for this to complete before continuing, you could create a Promise here instead by changing the return value of this function to `return new Promise(resolve => { ... } )` and then in the callback below, after `applyDomainContactValidationResults`, call `resolve()`.